tests/unittest_manager.py

```diff
@@ -184,41 +185,48 @@ def test_namespace_and_file_mismatch(self):
             sys.modules.pop("foogle")
 
     def _test_ast_from_zip(self, archive):
-        origpath = sys.path[:]
         sys.modules.pop("mypypa", None)
         archive_path = resources.find(archive)
         sys.path.insert(0, archive_path)
-        try:
-            module = self.manager.ast_from_module_name("mypypa")
-            self.assertEqual(module.name, "mypypa")
-            end = os.path.join(archive, "mypypa")
-            self.assertTrue(
-                module.file.endswith(end), f"{module.file} doesn't endswith {end}"
-            )
-        finally:
-            # remove the module, else after importing egg, we don't get the zip
-            if "mypypa" in self.manager.astroid_cache:
-                del self.manager.astroid_cache["mypypa"]
-                del self.manager._mod_file_cache[("mypypa", None)]
-            if archive_path in sys.path_importer_cache:
-                del sys.path_importer_cache[archive_path]
-            sys.path = origpath
+        module = self.manager.ast_from_module_name("mypypa")
+        self.assertEqual(module.name, "mypypa")
+        end = os.path.join(archive, "mypypa")
+        self.assertTrue(
+            module.file.endswith(end), f"{module.file} doesn't endswith {end}"
+        )
+
+    @contextmanager
+    def _restore_package_cache(self):
+        orig_path = sys.path[:]
```

Comment:
Is this equivalent to ``orig_path = list(sys.path)``?

Reply:
Functionally, yes. It's the idiom that I'm more used to for making a shallow copy of a list, but in this specific instance this line is copied from `_test_ast_from_zip` above.

Reply:
Thank you for the explanation :)